### PR TITLE
test: add legacy state order middleware unittests and refactor

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Based on Sasha's comment https://github.com/cowprotocol/cowswap/pull/2462#discussion_r1198624881
And based on Sasha's base test PRs ([1](https://github.com/cowprotocol/cowswap/pull/2530) and [2](https://github.com/cowprotocol/cowswap/pull/2531))

- Refactor legacy/state/orders/middleware
- Add unit tests
- [Bonus] Removed some unused components that had some broken tests

I'm sorry this is *very* verbose.
Focus on middleware/index and related subfiles

# To Test

1. Unit tests
2. Verify sound, popup and nps still work as before

### CODE REVIEW

File: src/legacy/state/orders/middleware/appziMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const order = getOrderByIdFromState(orders, orderId)?.order
+  const openSince = order?.openSince
+  const explorerUrl = getExplorerOrderLink(chainId, orderId)
+
+  // Open Appzi NPS for limit orders only if they were filled before `PENDING_TOO_LONG_TIME` since creating
+  if (order?.class === OrderClass.LIMIT && npsParams?.traded && isOrderInPendingTooLong(openSince)) {
+    return
+  }
+
+  openNpsAppziSometimes({ ...npsParams, secondsSinceOpen: timeSinceInSeconds(openSince), explorerUrl, chainId })
```
Comment (alfetopito) [inline src/legacy/state/orders/middleware/appziMiddleware.ts:52]: Not 100% sure this is correct for expiring limit orders, but I don't want to change behaviour in this PR.
Will update if needed in a follow up

File: src/legacy/state/orders/middleware/popupMiddleware.test.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      expect(batchPresignOrdersPopup).toHaveBeenCalledTimes(0)
+    })
+
+    it('should call batchFulfillOrderPopup when is batch fulfill order action', () => {
+      when(actionMock.type).thenReturn('order/fullfillOrdersBatch')
+
+      popupMiddleware(instance(mockStore))(nextMock)(instance(actionMock))
+
+      // TODO: can't get this to work, mockStore is never matched
+      // expect(batchFulfillOrderPopup).toHaveBeenCalledWith(null, { chainId: 1 }, 'mock orders')
```
Comment (alfetopito) [inline src/legacy/state/orders/middleware/popupMiddleware.test.ts:95]: Couldn't get this working. The first parameter doesn't match.
Open to suggestions

**Comment by elena-zh** [comment]
Hey @alfetopito , I'm still testing, however, I decided to report the 1st issue I have faced:
white screen is displayed when I place an  order (regular or ETH)
eth order placed

and when this order is filled (I understood it by the sound)=)
image

and when it is cancelled:
image
 Will proceed with Limit orders now

**Comment by elena-zh** [comment]
Limit orders page crashes as well on order placed/filled/expires events. 
Once I even have not heard the sound: 
limit filled order

**Review by elena-zh — APPROVED** [review]
Works fine now!

File: src/legacy/state/orders/middleware/index.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+//   document.body.classList.remove('lightning')
+// }
+//
+// function addLightningEffect() {
+//   document.body.classList.add('lightning')
+//
+//   setTimeout(() => {
+//     removeLightningEffect()
+//   }, 3000)
+// }
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply src/legacy/state/orders/middleware/index.ts:15]: Just delete them then

**Review by shoom3301 — APPROVED** [review]
This is amazing refactoring! You are a hero!
It also will help us to refactor this logic in the future.

**Review by anxolin — COMMENTED** [review]
this one, was too big too review. I think it was simplified in another PR though

**Pull request merged**